amazon/dynamodb/client/src/main/kotlin/org/http4k/connect/amazon/dynamodb/mapper/DynamoDbIndexMapper.kt

```diff
@@ -38,6 +38,21 @@ class DynamoDbIndexMapper<Document : Any, HashKey : Any, SortKey : Any>(
         .flatMap { result -> result.onFailure { it.reason.throwIt() } }
         .map(itemLens)
 
```

Comment:
Why do these need to be added to `DynamoDbIndexMapper`?  It seems to me they already exist as extensions functions in `DynamoDbQueryDsl.kt`

Reply:
I have moved these functions in this [commit](https://github.com/http4k/http4k-connect/pull/387/commits/8af9ad6f41af3f62d17cf66398d7b7b8c8f97e00). I had the impression the code gets more maintainable having all the scan/query/count functions together in the same class, and I didn't have to change the visibility of the `schema` property. But we can of course revert it. What would be your criteria to decide whether a function should be provided as a type member or as an extension (within the same library)?

Reply:
Ah I didn't know the originals had been removed.  I was worried they had been duplicated.

The way I see it, the DSL functions are a convenience layer built on top of the original interface, which in my book, means they should be extension functions.  However you make two good points:

- If extensions functions are not co-located, they are less maintainable (and discoverable)
- The `schema` property would need to be made public, which is fine

It would be my preference to keep the functions as extensions, but I trust you to make the decision you feel is best.  @daviddenton  this has my :shipit: unless Oliver wants to make another change.